Documentation/network/kubernetes/kubeproxy-free.rst

```diff
@@ -1617,6 +1617,13 @@ Limitations
     * The neighbor discovery in a multi-device environment doesn't work with the runtime device
       detection which means that the target devices for the neighbor discovery doesn't follow the
       device changes.
+    * When socket-LB feature is enabled, pods connected UDP traffic to services
+      can continue to send traffic to deleted service backend. Cilium agent
+      handles such scenarios by forcefully terminating sockets in the host network
+      namespaced that are connected to deleted backends, so that pods can be
+      load-balanced to active backends. This functionality requires these
+      kernel configs to be enabled ``CONFIG_INET_DIAG``, ``CONFIG_INET_UDP_DIAG``
```

Comment:
On ubuntu 22.04, kernel 5.19.0-50, `inet_diag` and `udp_diag` are kernel modules. It seems that those modules are loaded after destroying a UDP socket. Do you know what makes the kernel load the modules? The netlink requests that `socketDiagUDPExecutor` sends?

```
# Before
$ lsmod | head
Module                  Size  Used by
xt_TPROXY              20480  16
nf_tproxy_ipv6         20480  1 xt_TPROXY
nf_tproxy_ipv4         20480  1 xt_TPROXY
ip_set                 53248  0
xt_CT                  16384  32
xt_mark                16384  16
cls_bpf                24576  36
sch_ingress            16384  24
vxlan                 102400  0

# After destroying a socket
level=info msg="debug-aditi destroying connections to backend {10.244.1.184 {UDP 2399} 0}" subsys=service
level=info msg="debug-aditi check rev nat {38185 2399 10.244.3.92 10.244.1.184 0 [12647 0]} [12647 0]" subsys=service
level=info msg="Forcefully terminated sockets" errors="<nil>" failed=0 filter="{10.244.1.184 2399 2 17 0x25ea3c0}" subsys=datapath-sockets success=

$ lsmod | head
Module                  Size  Used by
udp_diag               16384  0
inet_diag              24576  1 udp_diag
xt_TPROXY              20480  16
nf_tproxy_ipv6         20480  1 xt_TPROXY
nf_tproxy_ipv4         20480  1 xt_TPROXY
ip_set                 53248  0
xt_CT                  16384  32
xt_mark                16384  16
cls_bpf                24576  36


# Ubuntu 22.04 kernel 5.19.0-50
$ cat /boot/config-$(uname -r) | grep INET
CONFIG_INET=y
CONFIG_INET_AH=m
CONFIG_INET_ESP=m
CONFIG_INET_ESP_OFFLOAD=m
CONFIG_INET_ESPINTCP=y
CONFIG_INET_IPCOMP=m
CONFIG_INET_TABLE_PERTURB_ORDER=16
CONFIG_INET_XFRM_TUNNEL=m
CONFIG_INET_TUNNEL=m
CONFIG_INET_DIAG=m
CONFIG_INET_TCP_DIAG=m
CONFIG_INET_UDP_DIAG=m
CONFIG_INET_RAW_DIAG=m
CONFIG_INET_DIAG_DESTROY=y
CONFIG_INET6_AH=m
CONFIG_INET6_ESP=m
CONFIG_INET6_ESP_OFFLOAD=m
CONFIG_INET6_ESPINTCP=y
CONFIG_INET6_IPCOMP=m
CONFIG_INET6_XFRM_TUNNEL=m
CONFIG_INET6_TUNNEL=m
CONFIG_INET_MPTCP_DIAG=m
CONFIG_NF_TABLES_INET=y
CONFIG_NFT_REJECT_INET=m
CONFIG_NFT_FIB_INET=m
CONFIG_NF_FLOW_TABLE_INET=m
CONFIG_INET_DCCP_DIAG=m
CONFIG_INET_SCTP_DIAG=m
CONFIG_USB_SERIAL_OMNINET=m
```

Reply:
I wasn't aware of this. Thanks!
Could you confirm if the modules were loaded on the first syscall invocation, and that the agent didn't receive `EOPNOTSUPP`?

Reply:
`inet_diag` and `udp_diag` were automatically loaded on the first syscall invocation, and the agent didn't receive `EOPNOTSUPP` on my environment.(Ubuntu 22.04, kernel 5.19)